readthedocs/search/documents.py

```diff
@@ -15,29 +14,31 @@
 page_index.settings(**page_conf['settings'])
 
 
-@project_index.doc_type
-class ProjectDocument(RTDDocTypeMixin, DocType):
+log = logging.getLogger(__name__)
 
-    class Meta(object):
-        model = Project
-        fields = ('name', 'slug', 'description')
-        ignore_signals = settings.ES_PROJECT_IGNORE_SIGNALS
 
+@project_index.doc_type
+class ProjectDocument(DocType):
+
+    # Metadata
     url = fields.TextField(attr='get_absolute_url')
     users = fields.NestedField(properties={
         'username': fields.TextField(),
         'id': fields.IntegerField(),
     })
     language = fields.KeywordField()
 
+    class Meta(object):
+        model = Project
+        fields = ('name', 'slug', 'description')
+        ignore_signals = True
+
     @classmethod
-    def faceted_search(cls, query, language=None, using=None, index=None):
+    def faceted_search(cls, query, user, language=None):
+        from readthedocs.search.faceted_search import ProjectSearch
         kwargs = {
-            'using': using or cls._doc_type.using,
-            'index': index or cls._doc_type.index,
-            'doc_types': [cls],
```

Comment:
@ericholscher I think we can pass the `doc_type` from here to avoide the lazy import.

Reply:
Will take a peek at that in a refactor, I think it's OK for now.